autopush/tests/test_rs_integration.py

```diff
@@ -950,102 +1086,159 @@ def test_webpush_monthly_rotation_prior_record_exists(self):
-        pk_hex = vapid['crypto-key']
-        chid = str(uuid.uuid4())
+    def test_webpush_monthly_rotation_no_channels(self):
+        from autopush.db import make_rotating_tablename
         client = Client("ws://localhost:{}/".format(self.connection_port))
         yield client.connect()
         yield client.hello()
-        yield client.register(chid=chid, key=pk_hex)
-
-        # Send an update with a properly formatted key.
-        yield client.send_notification(vapid=vapid)
+        yield client.disconnect()
 
-        # now try an invalid key.
-        new_key = ecdsa.SigningKey.generate(curve=ecdsa.NIST256p)
-        vapid = _get_vapid(new_key, claims)
+        # Move the client back one month to the past
+        last_month = make_rotating_tablename(
+            prefix=self.conn.conf.message_table.tablename, delta=-1)
+        yield deferToThread(
+            self.conn.db.router.update_message_month,
+            client.uaid,
+            last_month
+        )
 
-        yield client.send_notification(
-            vapid=vapid,
-            status=401)
+        # Verify the move
+        c = yield deferToThread(self.conn.db.router.get_uaid,
+                                client.uaid
+                                )
+        assert c["current_month"] == last_month
 
-        yield self.shut_down(client)
+        # Verify there's no channels
+        exists, chans = yield deferToThread(
+            self.conn.db.message.all_channels,
+            client.uaid,
+        )
+        assert exists is False
+        assert len(chans) == 0
 
-    @inlineCallbacks
-    def test_with_bad_key(self):
-        chid = str(uuid.uuid4())
-        client = Client("ws://localhost:{}/".format(self.connection_port))
+        # Connect the client, verify the migration
         yield client.connect()
         yield client.hello()
-        result = yield client.register(chid=chid, key="af1883%&!@#*(",
-                                       status=400)
-        assert result["status"] == 400
 
+        # Wait up to 2 seconds for the table rotation to occur
+        start = time.time()
+        while time.time()-start < 2:
+            c = yield deferToThread(
+                self.conn.db.router.get_uaid,
+                client.uaid,
+            )
+            if c["current_month"] == self.conn.db.current_msg_month:
+                break
+            else:
+                yield deferToThread(time.sleep, 0.2)
+
+        # Verify the month update in the router table
+        c = yield deferToThread(self.conn.db.router.get_uaid,
+                                client.uaid)
+        assert c["current_month"] == self.conn.db.current_msg_month
         yield self.shut_down(client)
 
 
-class TestRustWebPushBroadcast(unittest.TestCase):
-    connection_port = 9050
+class TestRustAndPythonWebPush(unittest.TestCase):
+    connection_port = 9052
     endpoint_port = 9060
-    router_port = 9070
+    router_port = 9072
```

Comment:
nit: same as above, offset these?

Reply:
Either way one needs to look at the ports at the top to determine if any collide, and the tests won't pass if they collide since the server will fail to start. Is there some reason for using offsets I'm missing?